operator/pkg/gateway-api/helpers/referencegrants.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package helpers
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	schema "k8s.io/apimachinery/pkg/runtime/schema"
+	gatewayv1beta1 "sigs.k8s.io/gateway-api/apis/v1beta1"
+)
+
+// isReferenceAllowed returns true if the reference is allowed by the reference grant.
```

Comment:
`isReferenceAllowed` -> `IsBackendReferenceAllowed`

Reply:
Will change this in the next PR where i also add checks for certificates